File: libaums/src/main/java/com/github/mjdev/libaums/ErrNo.java
package com.github.mjdev.libaums;

import android.util.Log;

/**
 * Created by magnusja on 5/19/17.
 */

public class ErrNo {

    private static String TAG = ErrNo.class.getSimpleName();
    private static boolean isInited = true;

    static {
        try {
            System.loadLibrary("errno-lib");
        } catch(UnsatisfiedLinkError e) {
            isInited = false;
            Log.e(TAG, "could not load errno-lib", e);
        }
    }


    public static int getErrno() {
        if(isInited) {
            return getErrnoNative();
        } else {
            return 1337;
        }
    }

    public static String getErrstr() {
        if(isInited) {
            return getErrstrNative();
        } else {
            return "errno-lib could not be loaded!";
        }
    }

    public static native int getErrnoNative();
    public static native String getErrstrNative();
}

File: libaums/src/main/java/com/github/mjdev/libaums/ErrNo.kt
package com.github.mjdev.libaums

import android.util.Log

/**
 * Created by magnusja on 5/19/17.
 */

object ErrNo {

    private val TAG = "ErrNo"
    private var isInited = true


    val errno: Int
        get() = if (isInited) {
            errnoNative
        } else {
            1337
        }

    val errstr: String
        get() = if (isInited) {
            errstrNative
        } else {
            "errno-lib could not be loaded!"
        }

    private val errnoNative: Int
        external get
    private val errstrNative: String
        external get

    init {
        try {
            System.loadLibrary("errno-lib")
        } catch (e: UnsatisfiedLinkError) {
            isInited = false
            Log.e(TAG, "could not load errno-lib", e)
        }

    }
}
